satellite/console/consoleweb/server.go

```diff
@@ -192,6 +209,282 @@ func (server *Server) appHandler(w http.ResponseWriter, r *http.Request) {
 	}
 }
 
+// authMiddlewareHandler performs initial authorization before every request
+func (server *Server) authMiddlewareHandler(handler http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		ctx := r.Context()
+		var err error
+		defer mon.Task()(&ctx)(&err)
+		token := getToken(r)
+
+		ctx = auth.WithAPIKey(ctx, []byte(token))
+		auth, err := server.service.Authorize(ctx)
+		if err != nil {
+			ctx = console.WithAuthFailure(ctx, err)
+		} else {
+			ctx = console.WithAuth(ctx, auth)
+		}
+
+		handler.ServeHTTP(w, r.Clone(ctx))
+	})
+}
+
+// tokenRequestHandler authenticates User by credentials and returns auth token
+func (server *Server) tokenRequestHandler(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+	var err error
+	defer mon.Task()(&ctx)(&err)
+
+	type tokenRequestModel struct {
+		Email    string `json:"email"`
+		Password string `json:"password"`
+	}
+
+	var tokenRequest tokenRequestModel
+	err = json.NewDecoder(r.Body).Decode(&tokenRequest)
+	if err != nil {
+		server.serveJsonError(w, 400, err)
+	}
+
+	token, err := server.service.Token(ctx, tokenRequest.Email, tokenRequest.Password)
+	if err != nil {
+		server.serveJsonError(w, 404, err)
+		return
+	}
+
+	err = json.NewEncoder(w).Encode(token)
+	if err != nil {
+		server.log.Debug("Error serializing response: " + err.Error())
+	}
+}
+
+// changeAccountPasswordRequestHandler updates password for a given user
+func (server *Server) changeAccountPasswordRequestHandler(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+	var err error
+	defer mon.Task()(&ctx)(&err)
+
+	type changePasswordRequestModel struct {
+		Password    string `json:"password"`
+		NewPassword string `json:"newPassword"`
+	}
+
+	var passwordChange changePasswordRequestModel
+	err = json.NewDecoder(r.Body).Decode(&passwordChange)
+	if err != nil {
+		server.serveJsonError(w, 400, err)
+		return
+	}
+
+	err = server.service.ChangePassword(ctx, passwordChange.Password, passwordChange.NewPassword)
+	if err != nil {
+		server.serveJsonError(w, 404, err)
+		return
+	}
+
+	w.WriteHeader(200)
+}
+
+// createNewUserRequestHandler gets password hash value and creates new inactive User
+func (server *Server) createNewUserRequestHandler(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+	var err error
+	defer mon.Task()(&ctx)(&err)
+
+	type createUserRequestModel struct {
+		console.CreateUser
+		Secret         string `json:"secret"`
+		ReferrerUserID string `json:"referrerUserId"`
+	}
+
+	var model createUserRequestModel
+	err = json.NewDecoder(r.Body).Decode(&model)
+	if err != nil {
+		server.serveJsonError(w, 400, err)
+		return
+	}
+
+	secret, err := console.RegistrationSecretFromBase64(model.Secret)
+	if err != nil {
+		server.log.Debug("register: ", zap.String("rawSecret", model.Secret))
```

Comment:
why do we need to log it?

Reply:
is it safe enough?

Reply:
I moved this log method from mutation layer